File: src/main/java/org/kite9/diagram/model/Label.java
package org.kite9.diagram.model;

import org.kite9.diagram.model.position.End;
import org.kite9.diagram.model.style.LabelPlacement;

/**
 * DiagramElement to contain a label for an edge, container or diagram.
 * Labels take up space on the diagram, so they have to be processed in the *orthogonalization* phase.
 * however they don't have connections so they are excluded from the Planarization phase.
 */
public interface Label extends Rectangular {
		
	public boolean isConnectionLabel();
	
	/**
	 * If this is a connection label, returns the end of the connection that it is for.
	 */
	public End getEnd();
	
	public LabelPlacement getLabelPlacement();
}

File: k9layout/src/main/kotlin/org/kite9/diagram/model/Label.kt
package org.kite9.diagram.model

import org.kite9.diagram.model.position.End
import org.kite9.diagram.model.style.HorizontalAlignment
import org.kite9.diagram.model.style.LabelPlacement
import org.kite9.diagram.model.style.VerticalAlignment

/**
 * DiagramElement to contain a label for an edge, container or diagram.
 * Labels take up space on the diagram, so they have to be processed in the *orthogonalization* phase.
 * however they don't have connections so they are excluded from the Planarization phase.
 */
interface Label : Rectangular {

    fun isConnectionLabel(): Boolean

    /**
     * If this is a connection label, returns the end of the connection that it is for.
     */
    fun getEnd(): End?

    fun getLabelPlacement(): LabelPlacement?

}